feat: add install script

iloveitaly opened the pull request:
I'm having fun with the free claude credits. Need to review + test this further, but I've used this script in other projects and it's worked great.

claude committed:
feat: add bash shell install script

Add a POSIX-compliant shell installation script for the tasker binary,
adapted from the localias project. This script provides a convenient way
to install tasker without requiring Go to be installed.

Features:
- Automatic platform and architecture detection
- Support for Linux, macOS, and Windows
- Automatic extraction of tar.gz/zip archives
- Version selection (latest or specific version)
- Custom installation directory
- Non-interactive mode for CI/CD environments
- Comprehensive error handling and user feedback

The script supports the following platforms and architectures:
- Linux: amd64, arm64, 386, armv6
- macOS (Darwin): amd64, arm64
- Windows: amd64

Also updated README.md with detailed installation instructions including
examples for various use cases.

Helpful for installing on GitHub Actions and other Linux environments.

claude committed:
fix: handle archive directory structure in install script

The tar.gz and zip archives contain a parent directory with the binary
inside. Updated extraction logic to:
- Use --strip-components=1 for tar.gz to extract without parent dir
- Move contents up one level for zip files to handle nested structure

Tested full installation process and verified binary works correctly.

claude committed:
docs: simplify install script options in README

Instead of listing all options inline, direct users to --help flag.
This reduces maintenance burden and keeps documentation in sync.

adhocore commented on install.sh:
```diff
@@ -0,0 +1,504 @@
+# Test if a location is writable by trying to write to it. Windows does not let
+# you test writability other than by writing: https://stackoverflow.com/q/1999988
+test_writeable() {
+  path="${1:-}/test.txt"
```
should we use random filename? so it won't remove the preexisting test.txt file later

adhocore commented on install.sh:
```diff
@@ -0,0 +1,504 @@
+      error "Error reading from prompt (please re-run with the '--yes' option)"
+      exit 1
+    fi
+    if [ "$yn" != "y" ] && [ "$yn" != "yes" ]; then
```
how about different alias/case of yes: YA, ya, ja, yea etc

so i think if first char of `$yn` in lowercase is `y` or `j` then it is positive

adhocore reviewed (commented):
thanks for PR. i ran the installer and it worked in my linux amd64. but how to verify in other flavors or OSes?

currently, since tasker is now a binary in `/usr/local/bin`,  `tasker -h` with below output seems a bit lacking:

```
Usage of tasker:
  -file string
        The task file in crontab format (without user)
  -out string
        The fullpath to file where output from tasks are sent to
  -shell string
        The shell to use for running tasks (default "/usr/bin/bash")
  -tz string
        The timezone to use for tasks (default "Local")
  -until int
        The timeout for task daemon in minutes
  -v    Show version
  -verbose
        The verbose mode outputs as much as possible
```

basically a short intro before usage would be great if that's possible natively

adhocore commented:
[feat: add bash shell install script](https://github.com/adhocore/gronx/pull/64/commits/2036ee1bddd3965ed53f6ce8db77534e6bcbee33)

i think this commit message prefix must be `build: ...` instead of `feat: ...`

adhocore commented:
closing because pr author isn't interested to put in any more effort than AI's readymade solution.

iloveitaly commented:
Just busy with other things :) I'll come back to this later, if that's ok.

adhocore commented:
that's not ok. but encouraged and welcomed. ;)

iloveitaly committed:
fix(install): use unique temp file and improve confirmation prompt

- Use .tasker_write_test_98087 instead of test.txt in test_writeable to avoid conflicts
- Allow case-insensitive and alternative yes/ja responses in confirm prompt

iloveitaly commented:
Made the requested updated! Seems to be working well!

adhocore commented:
thank you so much 👍 

(merging after detailed check asap)

adhocore reviewed (approved):
i didn't review the installer line-by-line but a quick run could install tasker so i think its good to go.

one little caveat, the installed tasker binary misses version `tasker -v` gives vn/a i.e version 'not/available' - it may or may not be possible to address that in here
